File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/util/NetHelper.java


package com.example.javademogithubpractice.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.annotation.NonNull;
import android.util.Log;

public enum NetHelper {
	INSTANCE;

	public static final int TYPE_DISCONNECT = 0;
	public static final int TYPE_WIFI = 1;
	public static final int TYPE_MOBILE = 2;


	private int mCurNetStatus;
	private Context mContext;

	public void init(Context context){
		mContext = context;
		checkNet();
	}

	public void checkNet(){
		try {
			ConnectivityManager connectivity = (ConnectivityManager) mContext
					.getSystemService(Context.CONNECTIVITY_SERVICE);
			if (connectivity != null) {
				NetworkInfo info = connectivity.getActiveNetworkInfo();
				if (info != null&& info.isAvailable()) {
					if (info.getState() == NetworkInfo.State.CONNECTED) {
						if(info.getType() == ConnectivityManager.TYPE_WIFI)
							mCurNetStatus =  TYPE_WIFI;
						if(info.getType() == ConnectivityManager.TYPE_MOBILE)
							mCurNetStatus =  TYPE_MOBILE;
					}
				} else{
					mCurNetStatus = TYPE_DISCONNECT;
				}
			}
		} catch (Exception e) {
			Log.v("error",e.toString());
			e.printStackTrace();
			mCurNetStatus = TYPE_DISCONNECT;
		}
	}

	@NonNull
    public Boolean getNetEnabled(){
		return mCurNetStatus == TYPE_MOBILE || mCurNetStatus == TYPE_WIFI;
	}

	@NonNull
    public Boolean isMobileStatus(){
		return mCurNetStatus == TYPE_MOBILE;
	}

	public int getNetStatus() {
		return mCurNetStatus;
	}
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/util/NetHelper.kt
package com.example.kotlindemogithubproject.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.util.Log
import androidx.annotation.NonNull

enum class NetHelper {
    INSTANCE;


    var netStatus: Int = 0
        private set
    private var mContext: Context? = null

    val netEnabled: Boolean
        @NonNull
        get() = netStatus == TYPE_MOBILE || netStatus == TYPE_WIFI

    val isMobileStatus: Boolean
        @NonNull
        get() = netStatus == TYPE_MOBILE

    fun init(context: Context) {
        mContext = context
        checkNet()
    }

    fun checkNet() {
        try {
            val connectivity = mContext!!
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            if (connectivity != null) {
                val info = connectivity.activeNetworkInfo
                if (info != null && info.isAvailable) {
                    if (info.state == NetworkInfo.State.CONNECTED) {
                        if (info.type == ConnectivityManager.TYPE_WIFI)
                            netStatus = TYPE_WIFI
                        if (info.type == ConnectivityManager.TYPE_MOBILE)
                            netStatus = TYPE_MOBILE
                    }
                } else {
                    netStatus = TYPE_DISCONNECT
                }
            }
        } catch (e: Exception) {
            Log.v("error", e.toString())
            e.printStackTrace()
            netStatus = TYPE_DISCONNECT
        }

    }

    companion object {

        val TYPE_DISCONNECT = 0
        val TYPE_WIFI = 1
        val TYPE_MOBILE = 2
    }
}
